test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2053,6 +2030,14 @@ protected static IndexVersion minimumIndexVersion() throws IOException {
         return minVersion;
     }
 
+    private static int parseLegacyVersion(String version) {
+        Matcher semanticVersionMatcher = Pattern.compile("^(\\d+\\.\\d+\\.\\d+)\\D?.*").matcher(version);
```

Comment:
The Pattern should probably be pulled out into a static field so it can be reused

Reply:
Yeah, I was conflicted about it; I wanted this to be self contained, but on the other hand it's silly to re-create it every time. I will extract it.